plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1219,35 +1256,29 @@ public void testFileSortingWithLargerTable()
         dropTable(tableName);
     }
 
+    @Test(dataProvider = "sortedTableWithQuotedIdentifierCasing")
+    public void testCreateSortedTableWithQuotedIdentifierCasing(String columnName, String sortField)
+    {
+        String tableName = "sorting_" + randomNameSuffix();
+        assertUpdate(format("CREATE TABLE %s (%s bigint) WITH (sorted_by = ARRAY['%s'])", tableName, columnName, sortField));
+        dropTable(tableName);
+    }
+
     @DataProvider(name = "sortedTableWithQuotedIdentifierCasing")
     public static Object[][] sortedTableWithQuotedIdentifierCasing()
     {
         return new Object[][] {
-                {"x", "x", true},
-                {"X", "x", true},
-                {"\"x\"", "x", true},
-                {"\"X\"", "x", true},
-                {"x", "\"x\"", true},
-                {"X", "\"x\"", true},
-                {"\"x\"", "\"x\"", true},
-                {"\"X\"", "\"x\"", true},
+                {"x", "x"},
+                {"X", "x"},
+                {"\"x\"", "x"},
+                {"\"X\"", "x"},
+                {"x", "\"x\""},
+                {"X", "\"x\""},
+                {"\"x\"", "\"x\""},
+                {"\"X\"", "\"x\""},
```

Comment:
The test coverage is missing negative test cases.
What is rejected today, what will be supported in the future?

Also, using `COL` or `NAME` instead of `X` would make it easier to read.
Small x and uppercase X are quite similar, especially when decorated with many quotation marks. and backslashes.